uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/leave-request-api_test.js

```diff
@@ -304,6 +304,26 @@ define([
       });
     });
 
+    describe('delete()', function () {
+      var idToDelete = '123';
+
+      beforeEach(function () {
+        spyOn(LeaveRequestAPI, 'sendGET');
```

Comment:
Yep exactly. If you look at the tests that i simply removed, those were tests that were just checking the structure of the endpoint response. But it doesn't make sense to test for that, it's not the responsibility of the API service to define the structure of the response.
```js
// example
it('has the expected structure', function () {
  promise.then(response) {
    // the structure does not depend on the api service itself, what's the point of testing it?
    expect(response.foo).toBeDefine();
    expect(response.bar).toBeDefine();
  });
});
```
instead, if an api perform some transformation on the endpoint response before returning the value, then that is something worth testing
```js
it('has the expected structure', function () {
  promise.then(response) {
    // the structure does not depend on the api service itself, what's the point of testing this?
    expect(response.someCustomField).toBeDefined();
    expect(response.someCustomField).toBe(/* some expected value */);
  });
});
```